packages/core/src/form-field/FormFieldLabel.tsx

```diff
@@ -15,14 +15,20 @@ export const FormFieldLabel = ({
   children,
   ...restProps
 }: Omit<TextProps<"label">, "variant" | "styleAs">) => {
-  const { a11yProps, disabled } = useFormFieldProps();
+  const { a11yProps, disabled, necessity } = useFormFieldProps();
   const targetWindow = useWindow();
   useComponentCssInjection({
     testId: "salt-form-field-label",
     css: formFieldLabelCss,
     window: targetWindow,
   });
 
+  const necessityLabel = necessity
```

Comment:
I think it can't be compositional given label and control both need to know about the necessity, for me it makes sense to be added to the context

> Will this impact potential localization need (asked from UITK a few times), where the display is tied to prop?

what do you mean by this exactly?

Reply:
If someone needed a different language for "required" display label

Reply:
ah gotcha! thank you, will raise with Ben